playwright-e2e/dsm/pages/participant-list-page.ts

```diff
@@ -245,88 +245,94 @@ export default class ParticipantListPage extends DsmPageBase {
-            break;
+          if (medicalRecord && participantID) {
+            shortID = JSON.stringify(value.esData.profile.hruid).replace(/['"]+/g, '');
+            if (!rgpProbandTab) {
+              // Do not have to find RGP Proband tab
+              logInfo(`Found participant with Short ID: ${shortID} to have tab: ${tab}`);
+              return shortID;
+            }
           }
         }
 
         if (rgpProbandTab) {
-          //If the participant has participantData, this seems to mean there's a proband tab in the account
-          const participantData = value.participantData[0];
-          if (participantData === undefined) {
-            continue; //Participant does not have a proband tab for some reason, skip them
-          }
-          firstName = value.esData.profile?.firstName;
-          if (!firstName) {
+          // Data in participantData seems to mean there's a proband tab
+          const participantData = value.participantData;
+          if (!participantData || participantData[0] === undefined) {
             continue;
           }
-          unformattedFirstName = JSON.stringify(firstName);
-          if (participantData && (firstName === testParticipantFirstName)) {
-            foundShortID = JSON.stringify(value.esData.profile.hruid).replace(/['"]+/g, '');
-            logInfo(`Found the RGP participant ${foundShortID} to have a proband tab`);
-            break;
-          }
+          shortID = JSON.stringify(value.esData.profile.hruid).replace(/['"]+/g, '');
+          logInfo(`Found RGP participant with Short ID: ${shortID} to have proband tabs`);
+          return shortID;
         }
+      } // end of for ...entries()
 
-        //Go to the next page if none of the participants in the current page are relevant
-        if (index === (amountOfParticipantsDisplayed - 1)) {
-          const hasNextPage = await this._table.paginator.hasNext();
-          if (hasNextPage) {
-            const [nextPageResponse] = await Promise.all([
-              waitForResponse(this.page, {uri: 'ui/filterList'}),
-              this._table.nextPage()
-            ]);
-            responseJson = JSON.parse(await nextPageResponse.text());
-          } else {
-            if (tab) {
-              throw new Error(`Could not find a participant with the ${tab} tab`);
-            }
-            if (rgpProbandTab) {
-              throw new Error(`Could not find a participant with the RGP Proband tab`);
-            }
-          }
-        }
+      // Get new /filterList response by go to the next page
+      const hasNext = await this._table.paginator.hasNext();
+      if (!hasNext || Date.now() > endTime) {
+        break; // no more participants or exceeded max timeout
       }
-    } while (foundShortID === '');
-    return foundShortID;
+
+      // continue with finding
+      const nextPageResponse = await this._table.nextPage();
+      responseJson = JSON.parse(await nextPageResponse.text());
```

Comment:
fetch and save new `/filterList` response should be outside of `for (const [index, value] of [...responseJson.participants].entries())`, not inside of it.